### ISSUE TITLE
Lint `return()` usage with pipe in `return_linter()`? 

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2548]
Ref. https://magrittr.tidyverse.org/news/index.html?q=return#breaking-changes-2-0-0

<img width="800" alt="Screenshot 2024-04-02 at 20 44 49" src="https://github.com/r-lib/lintr/assets/11330453/7f81cb32-0371-4cd5-bf34-31c86e1fbf00">

But this code currently doesn't lint:

``` r
library(lintr)
library(magrittr)

'my_function <- function() {
  "value" %>% identity() %>% return() # should produce lint
  "wrong value"
}' -> code

lint(
  text = code,
  linters = all_linters()
)
```

<sup>Created on 2024-04-02 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
that's `pipe_return_linter()`:

https://github.com/r-lib/lintr/blob/main/R/pipe_return_linter.R

**Comment by IndrajeetPatil** [comment]
I had a vague memory that we had something like this, but couldn't find it! 😅

So I guess the example above is a false negative?

**Comment by MichaelChirico** [comment]
Hmm, I do get a lint. Can you check you're on the latest `main`? I'll close for now:

```
<text>:2:30: warning: [pipe_return_linter] Avoid return() as the final step of a magrittr pipeline.  Instead, assign the output of the pipeline to a well-named object and return that.
  "value" %>% identity() %>% return() # should produce lint
                             ^~~~~~~~
```

**Comment by IndrajeetPatil** [comment]
Yes, that was it: I had somehow managed locally to regress to some old version of lintr. Thanks!
